### ISSUE TITLE
Observe audio widget property values from Python

### ISSUE THREAD

**Issue by benbovy** [benbovy/ipytone#42]
This is somewhat complementary to #30.

It would be nice if we could somehow observe from Python the audio widget properties that are usually updated at the audio sample rate. It doesn't make sense to observe those widgets at that high-frequency, but we could observe it at a lower resolution in different ways.

In the examples below only the widget property value is synchronized with the Python side, but we may also want to synchronize the Tone (transport and/or context) time at which those values have been sampled if we want to make something useful on the Python side. Synchronization latency is highly unpredictable and can greatly differ from one Jupyter deployment to another.

### Property values synced at defined regular intervals

We could imagine something like:

```python
def callback(change):
    with output:
        print(change['new'])

obs_id = audio_signal.schedule_observe(callback, names='value', time=0, interval=None, transport=True)
```

`callback` would be called when `ipytone.transport.start(0)`, and possibly called again at regular intervals. On the JS side, this would execute something like:

```typescript
transport_model.schedule((time) => {
  this.set('value', this.node.getValueAtTime(time), { 'silent': true });
  this.save_changes();
}, 0);
```

If `transport=False`, the current time of the main audio context is used, e.g.,

```python
obs_id = audio_signal.schedule_observe(callback, names='value', interval=1, transport=False)
```

will execute

```typescript
setInterval(() => {
  this.set('value', this.node.getValueAtTime(Tone.now()), { 'silent': true });
  this.save_changes();
}, 1000);
```

The returned value `obs_id` may be used to stop the scheduled observe: `audio_signal.clear_observe(obs_id)`.

### Property values synced conditionally

Trigger `callback` on the Python side only if some condition is met (e.g., for a `Meter` node when the RMS audio signal crosses a given threshold). This still needs some scheduled events to check the condition but it can be made at a much higher resolution on the front-end side (at the risk of streaming a lot of data frontend->backend if the condition is often met).

```typescript
setInterval((threshold) => {
  const value = this.node.getValue();
  if (value > threshold) {
    this.set('value', value, { 'silent': true });
    this.save_changes();
  }
}, 1000, threshold);
```

### Property values synced at scheduled events

~~This may be a bit more tricky. Maybe we can keep track of all scheduled events and hence also schedule setting model attributes and call `this.save_changes`. Or maybe there's an easy way to listen to values changes for Tone object attributes, e.g., using [Proxy](https://stackoverflow.com/a/52529719/2814564)?~~

We could reuse Tone.js events for more advanced sampling schemes, e.g.,

```typescript
const part = new Tone.Part(((time, value) => {
  this.set('value', this.node.getValueAtTime(time), { 'silent': true });
  this.save_changes();
  // the note and velocity are not useful here
}), [{ time: 0, note: "C1", velocity: 0 }, { time: "0:2", note: "C1", velocity: 0 }
]).start(0);
```

**Comment by benbovy** [comment]
xref #8

**Comment by benbovy** [comment]
Closed in #70.
